db/bus_test.go

```diff
@@ -0,0 +1,597 @@
+                    Address:  "addr",
+                    Version:  "v1.1",
+                    Hidden: true,
+                    LastSeenAt: 3,
+                    LastCheckedAt: 4,
+                    LastError: "nil",
+                    Status: "bad",
+                }, "*")
+            })
+
+            It("should receive a update-object message bus event for Agent, eventually", func(done Done) {
+                /* this is executed in a goroutine */
+                var a Agent
+
+                /* Update Agent object message bus event*/
+                receive(<-events, "update-object", "agent", &a)
+                Ω(a.UUID).Should(Equal("foo"))
+                Ω(a.Name).Should(Equal("agent"))
+                Ω(a.Address).Should(Equal("addr"))
+                Ω(a.Version).Should(Equal("v1.1"))
+                Ω(a.Hidden).Should(Equal(true))
+                Ω(a.LastSeenAt).Should(Equal(int64(3)))
+                Ω(a.LastCheckedAt).Should(Equal(int64(4)))
+                Ω(a.LastError).Should(Equal("nil"))
+                Ω(a.Status).Should(Equal("bad"))
+
+                close(done)
+            }, 2 /* timeout (in seconds) */)
+        })
+
+        /* Message bus tests for Job */
+        Context("when sending a createObject message for Job", func() {
+            BeforeEach(func() {
+                /* Create Job object */
+                db.sendCreateObjectEvent(Job{
+                    UUID:     "foo",
+                    Name:     "daily",
+                    Summary:  "A Daily Backup",
+                    KeepN:    2,
+                    KeepDays: 2,
+                    Schedule: "daily",
+                    Paused:   true,
+                    FixedKey: false,
+                }, "*")
+            })
+
+            It("should receive a create-object message bus event for Job, eventually", func(done Done) {
+                /* this is executed in a goroutine */
+                var j Job
+
+                /* Create job object bus event*/
+                receive(<-events, "create-object", "job", &j)
+                Ω(j.UUID).Should(Equal("foo"))
+                Ω(j.Name).Should(Equal("daily"))
+                Ω(j.Summary).Should(Equal("A Daily Backup"))
+                Ω(j.KeepN).Should(Equal(2))
+                Ω(j.KeepDays).Should(Equal(2))
+                Ω(j.Schedule).Should(Equal("daily"))
+                Ω(j.Paused).Should(Equal(true))
+                Ω(j.FixedKey).Should(Equal(false))
+
+                close(done)
+            }, 2 /* timeout (in seconds) */)
+        })
+
+        Context("when sending an updateObject message for Job", func() {
+            BeforeEach(func() {
+                /* Update Job object*/
+                db.sendUpdateObjectEvent(Job{
+                    UUID:     "foo",
+                    Name:     "weekly",
+                    Summary:  "A Weekly Backup",
+                    KeepN:    3,
+                    KeepDays: 3,
```

Comment:
I would set different values here, to make sure we aren't confusing KeepN / KeepDays